arviz/tests/external_tests/test_data_pymc.py

```diff
@@ -397,3 +397,36 @@ def test_no_model_deprecation(self):
         }
         fails = check_multiple_attrs(test_dict, inference_data)
         assert not fails
+
+    @pytest.mark.parametrize("save_warmup", [False, True])
+    def test_save_warmup(self, save_warmup):
+        with pm.Model():
+            pm.Uniform("u1")
+            pm.Normal("n1")
+            trace = pm.sample(
+                tune=100,
+                draws=200,
+                chains=2,
+                cores=1,
+                step=pm.Metropolis(),
+                discard_tuned_samples=False,
+            )
+            assert isinstance(trace, pm.backends.base.MultiTrace)
+            idata = from_pymc3(trace, save_warmup=save_warmup)
+        prefix = "" if save_warmup else "~"
+        test_dict = {
+            "posterior": ["u1", "n1"],
+            "sample_stats": ["~tune", "accept"],
+            f"{prefix}_warmup_posterior": ["u1", "n1"],
+            f"{prefix}_warmup_sample_stats": ["~tune"],
+            "~_warmup_log_likelihood": [],
+            "~log_likelihood": [],
+        }
+        fails = check_multiple_attrs(test_dict, idata)
+        assert not fails
+        assert idata.posterior.sizes["chain"] == 2
+        assert idata.posterior.sizes["draw"] == 200
+        if save_warmup:
+            # pylint: disable=protected-access
+            assert idata._warmup_posterior.dims["chain"] == 2
+            assert idata._warmup_posterior.dims["draw"] == 100
```

Comment:
- Just out of curiosity: why is the attribute to access different for `posterior` (`sizes`) and `_warmup_posterior` (`dims`), whereas you want to see the number of chains and draws in both cases?
- Also, does it mean we keep the `_` before `warmup`?

Reply:
_ should stay before warmup. it is not a mcmc draw (in hmc; maybe) so it is only for advanced users

Reply:
Regarding first question, both attributes do the same, the test Michael sent had sizes and I used dims to make it more xarray-y but missed half of them :shrug:

Reply:
@ahartikainen how opposed would you be to removing the `_` before warmup? Even if only for advanced users it is still for users, the default would still be `data.save_warmup = false` so unless they change it they would not have warmup groups, I feel like hiding them so much makes harder the live of advanced users without much gain for novices.

Reply:
Not too opposed.